Add docs for telemetry configuration (#15182)
DataDog/documentation: content/en/tracing/trace_collection/library_config/go.md

: **Default**: `null` <br>
Dynamically rename services through configuration. Services can be separated by commas or spaces, for example: `mysql:mysql-service-name,postgres:postgres-service-name`, `mysql:mysql-service-name postgres:postgres-service-name`.

`DD_INSTRUMENTATION_TELEMETRY_ENABLED`
: **Default**: `false` <br>

> Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. 
> When false, this telemetry data will not be collected.

Reviewer: ```suggestion
Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. When false, this telemetry data will not be collected.
```

: **Default**: `null` <br>
Dynamically rename services through configuration. Services can be separated by commas or spaces, for example: `mysql:mysql-service-name,postgres:postgres-service-name`, `mysql:mysql-service-name postgres:postgres-service-name`.

`DD_INSTRUMENTATION_TELEMETRY_ENABLED`
: **Default**: `false` <br>

> Datadog may collect [some telemetry][7]. This variable may be changed to default `true` in the future. To disable this telemetry collection, explicitly set this variable to `false`.

Reviewer 2: ```suggestion
Datadog may collect [environmental and diagnostic information about your system][7] to improve the product. 
When false, this telemetry data will not be collected.
```

Reviewer 2: If/when it is turned on by default in the future, that should go in release notes, and we should inform all customers in advance. We can update the docs at that time as well.

[4]: /tracing/trace_pipeline/ingestion_mechanisms/?tab=go#pagetitle
[5]: /tracing/advanced/setting_primary_tags_to_scope/#environment
[6]: /getting_started/tracing/#environment-name

> [7]: /tracing/configure_data_security/?tab=mongodb#telemetry-collection

Reviewer 2: ```suggestion
[7]: /tracing/configure_data_security#telemetry-collection
```
